src/vs/workbench/contrib/terminal/browser/terminalBaseContextualActions.ts

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IAction } from 'vs/base/common/actions';
+import { localize } from 'vs/nls';
+import { IOpenerService } from 'vs/platform/opener/common/opener';
+import { ContextualMatchResult, ITerminalContextualActionOptions, ITerminalInstance } from 'vs/workbench/contrib/terminal/browser/terminal';
+import { ITerminalCommand, ITerminalProcessManager } from 'vs/workbench/contrib/terminal/common/terminal';
+
+
+export const GitSimilarOutputRegex = /most similar command is\s+([^\s]{3,})\s*/;
+export const FreePortOutputRegex = /address already in use\s*(\d\.\s*){3}(\d\s*):((\d\s*){3}\d)\s*/;
```

Comment:
This is a bit hard to read, I think `\d\.\d\.\d\.\d` would be easier to understand for an IP. Also I don't understand the \s here: `(\d\s*){3}\d`

Reply:
I noticed that the line can wrap mid stream like

```
https://1.2.3.4:0
555
```

Reply:
maybe that's only because of how i've written the test though?

Reply:
for example, didn't work here bc
`a
ddress`
![Screen Shot 2022-09-21 at 9 46 27 AM](https://user-images.githubusercontent.com/29464607/191563749-a516784a-0a43-4f82-a665-a4f734cdec2a.png)

Reply:
so going to [].join('\s+') between all to avoid all of these possibilities

Reply:
oh this is quite complicated bc really need to join each of the chars in a word with `\s*`

Reply:
added tests and improved regex